iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/thread.hpp

```diff
@@ -34,8 +35,8 @@ constexpr uint64_t MAX_THREAD_NAME_LENGTH = 15U;
 using ThreadName_t = string<MAX_THREAD_NAME_LENGTH>;
 
 /// @todo iox-#1365 remove free functions
-void setThreadName(iox_pthread_t thread, const ThreadName_t& name) noexcept;
-ThreadName_t getThreadName(iox_pthread_t thread) noexcept;
+void setThreadName(std::thread::native_handle_type thread, const ThreadName_t& name) noexcept;
+ThreadName_t getThreadName(std::thread::native_handle_type thread) noexcept;
```

Comment:
The function is always passed a `t.native_handle()` where `t` is a `std::thread` instance. For example here: https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_hoofs/include/iceoryx_hoofs/internal/concurrent/periodic_task.inl#L63

It is not guaranteed that the `std::thread`-underlying implementation-defined thread handle is a pthread handle. In fact, with the FreeRTOS build, we are using an external C++ threading library (because bare metal C++ doesnt have `std::thread`) which has a different `std::thread` thread handle.

Reply:
Okay, I see. The problem is that we are in a transitional phase from `std::thread` to `iox::thread`. Have you tried to set the `iox_pthread_t` alias in the platform abstraction or does this also cause some issues?

If it does not work via the platform abstraction then we can of course apply this change.

Reply:
Okay, then lets keep it. This can be changed when we get rid of `std::thread`